docs/README.md

```diff
@@ -1,47 +1,190 @@
+   ```bash
+   docker pull stackrox/kube-linter
+   ```
+1. Add current directory with your `yaml` file as a read only volume for the
+   `docker run` command:
+   ```bash
+   docker run -v ${PWD}:/tmp:ro --rm -i stackrox/kube-linter lint /tmp/pod.yaml
+   ```
+
+## Building from source
+
+> [!NOTE] Before you build, make sure that you have [installed Go](https://golang.org/doc/install).
+
+To build KubeLinter from source:
+
+1. Clone the KubeLinter repository:
+   ```bash
+   git clone git@github.com:stackrox/kube-linter.git
+   ```
+1. Compile the source code:
+   ```bash
+   make build
+   ```
+   This command compiles the source code and creates `kube-linter` binary files
+   for each platform in the `.gobin` folder.
+1. Verify that the compiled binary is working:
+   ```bash
+   .gobin/kube-linter version
+   ```
+
+## Usage
+
+<!-- tabs:start -->
+
+#### ** Kubernetes **
+
+1. Consider the following sample pod specification file `pod.yaml`:
+   ```yaml
+   apiVersion: v1
+   kind: Pod
+   metadata:
+     name: security-context-demo
+   spec:
+     securityContext:
+       runAsUser: 1000
+       runAsGroup: 3000
+       fsGroup: 2000
+     volumes:
+     - name: sec-ctx-vol
+       emptyDir: {}
+     containers:
+     - name: sec-ctx-demo
+       image: busybox
+       resources:
+         requests:
+           memory: "64Mi"
+           cpu: "250m"
+       command: [ "sh", "-c", "sleep 1h" ]
+       volumeMounts:
+       - name: sec-ctx-vol
+         mountPath: /data/demo
+       securityContext:
+         allowPrivilegeEscalation: false
+   ```
+   > [!NOTE] This sample file has two production readiness issues and one
+   > security issue.
+   >
+   > **Security issue**
+   > - The container in this pod is not running as a read-only file system,
+   >   allowing it to write to the root filesystem.
+   > 
+   > **Production readiness issue**
+   > - The configuration doesn't specify the container's CPU requests and
+   >   limits, allowing it to consume excessive CPU. 
+   > - The configuration doesn't specify the container's memory requests and
+   >   limits, allowing it to consume excessive memory.
+
+1. To lint this file with KubeLinter, run the following command:
+   ```bash
+   kube-linter lint pod.yaml
+   ```
+1. KubeLinter runs the default checks and reports errors.
+   ```
+   pod.yaml: (object: <no namespace>/security-context-demo /v1, Kind=Pod) container "sec-ctx-demo" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)
+
+   pod.yaml: (object: <no namespace>/security-context-demo /v1, Kind=Pod) container    "sec-ctx-demo" has cpu limit 0 (check: unset-cpu-requirements, remediation: Set    your container's CPU requests and limits depending on its requirements. See    https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/   #requests-and-limits for more details.)
```

Comment:
Can you fix the spacing here (and below)? Seems like there are some places where there are multiple spaces in the output, but there shouldn't be.